api/envoy/config/route/v3/route_components.proto

```diff
@@ -403,6 +419,9 @@ message RouteMatch {
     // exactly match the *:path* header once the query string is removed.
     string path = 2;
 
+    // [#not-implemented-hide:]
+    ConnectConfig connect_matcher = 3;
```

Comment:
Does this belong as a path specifier or some other part of the match configuration? It seems not super logical to have it here, unless we want to say "the only situation in which an empty path can ever occur is CONNECT and this is effectively the empty path matcher". Doesn't seem future proof but I am happy to be convinced otherwise.